src/Analyzer/Pass/Expression/FinalStaticUsage.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Context;
+use PHPSA\Definition\ClassDefinition;
+
+class FinalStaticUsage implements AnalyzerPassInterface
+{
+    /**
+     * @param Expr\StaticCall $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr\StaticCall $expr, Context $context)
+    {
+        $classObject = $context->scope->getPointer()->getObject();
```

Comment:
I believe the scopePointer didn't contain the class so I couldn't check if the class was final, eventually I found it in the scope.